modules/sqs/main.tf

```diff
@@ -53,17 +53,81 @@ resource "aws_sqs_queue" "metadata_queue" {
+  visibility_timeout_seconds = 1800 # Set to double lambda max time
+  depends_on = [
+    aws_sqs_queue.staged_recovery_deadletter_queue
+  ]
+  redrive_policy = jsonencode({
+    deadLetterTargetArn = aws_sqs_queue.staged_recovery_deadletter_queue.arn,
+    maxReceiveCount     = 3 #if number of delayed messages in the queue is greater than this number, then the messages will be sent to its DLQ.
+  })
 }
 
+resource "aws_sqs_queue" "staged_recovery_deadletter_queue" {
+  name                       = "${var.prefix}-staged-recovery-deadletter-queue"
+  delay_seconds              = var.sqs_delay_time_seconds
+  max_message_size           = var.sqs_maximum_message_size
+  message_retention_seconds  = var.staged_recovery_queue_message_retention_time_seconds
+  tags                       = local.tags
+  visibility_timeout_seconds = 1800 # Set to same time as the staged recovery queue.
+}
+
+resource "aws_sqs_queue_policy" "deadletter_queue_policy" {
+  queue_url = aws_sqs_queue.staged_recovery_deadletter_queue.id
+  policy    = data.aws_iam_policy_document.staged_recovery_deadletter_queue_policy.json
+}
+
+data "aws_iam_policy_document" "staged_recovery_deadletter_queue_policy" {
+  statement {
+    effect    = "Allow"
+    resources = [aws_sqs_queue.staged_recovery_deadletter_queue.arn]
+    actions = [
+      "sqs:ChangeMessageVisibility",
+      "sqs:DeleteMessage",
+      "sqs:GetQueueAttributes",
+      "sqs:GetQueueUrl",
+      "sqs:ListQueueTags",
+      "sqs:ReceiveMessage",
+      "sqs:SendMessage",
+    ]
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "staged_recovery_deadletter_alarm" {
+  alarm_name          = "${aws_sqs_queue.staged_recovery_deadletter_queue.name}-not-empty-alarm"
+  alarm_description   = "Items are on the ${aws_sqs_queue.staged_recovery_deadletter_queue.name} queue"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = 1
+  metric_name         = "ApproximateNumberOfMessagesVisible"
+  namespace           = "AWS/SQS"
+  period              = 300 #In seconds. Valid values for period are 1, 5, 10, 30, or any multiple of 60. 
+  statistic           = "Average"
+  threshold           = 1 #alarm will be triggered if number of messages in the DLQ equals to this threshold.
+  treat_missing_data  = "notBreaching"
+  alarm_actions       = [aws_sns_topic.staged_recovery_dlq_alarm_topic.arn]
+  tags                = local.tags
+  dimensions = {
+    "QueueName" = aws_sqs_queue.staged_recovery_deadletter_queue.name
+  }
+}
+
+# SNS topic needed for cloudwatch alarm
+resource "aws_sns_topic" "staged_recovery_dlq_alarm_topic" {
+  name = "staged_recovery_deadletter_queue_alarm_topic"
+}
+
+resource "aws_sns_topic_subscription" "sns_subscription_for_alarm" {
+  depends_on = [aws_sns_topic.staged_recovery_dlq_alarm_topic]
+  topic_arn  = aws_sns_topic.staged_recovery_dlq_alarm_topic.arn
+  protocol   = "email"
+  endpoint   = "rhassan@contractor.usgs.gov"
```

Comment:
I am guessing it will be an “email” protocol. If so, TBD on who to add to the email.

Reply:
Could see an argument that these should come directly to us. Could also see an argument that this should be configurable, so the user can get a notification that something they tried failed. If the former, we should use an email address that targets our group. If the later, should add configuration.

Reply:
In the short term, I would make the endpoint email a variable so end users can add whatever email/group email they want. We will want to bring this up with the working group and discuss how they want to handle DLQ items and expectations. We may want to even consider other items like can we retry the DLQ messages (retryable error) and if the message cannot be retried, update our database with the error information for job status and delete the message as possible future work.

Reply:
so this will be a new required orca variable, right? @bhazuka

Reply:
added a new required orca variable for the email